### ISSUE TITLE
New rule proposal: Curly braces must not be omitted when parent expression spans multiple lines

### ISSUE THREAD

**Issue by alxyr** [DotNetAnalyzers/StyleCopAnalyzers#1488]
**Category**: Layout
**Rule**: Curly braces must not be omitted when parent expression spans multiple lines.
**Rationale**: Improve readability and maintainability of the code (when SA1503 is off).
**See also**: SA1503:CurlyBracketsMustNotBeOmitted and SA1519:CurlyBracketsMustNotBeOmittedFromMultiLineChildStatement

Bad:
```C#
if (alpha == beta &&
    beta == gamma)
    alpha = delta;
```
Good:
```C#
if (alpha == beta &&
    beta == gamma)
{
    alpha = delta;
}
```

**Comment by sharwell** [comment]
I think I'd be more in favor of modifying rule SA1519 to perform a combination of this and what it currently does. Something like "Curly braces must not be omitted when clause spans more than two lines". In addition to the case above, this would finally cover the following case:

```csharp
do
    Method(); // SA1519
while (condition);
```

**Comment by alxyr** [comment]
It would be ok to extend SA1519 but I'm not sure that this should affect the do-while loops -- short do-whiles will get two lines longer then:
```C#
do
{
    Method();
}
while (condition);
```
And sometimes there's a need to just transform a while-do loop to a do-while loop. With the new behavior, moving the loop condition to the end of the loop would require adding {} to the code (and a separate line for each of the braces).

**Comment by jnm2** [comment]
@sharwell So `SA1519:CurlyBracketsMustNotBeOmittedFromMultiLineChildStatement` would change to `CurlyBracketsMustNotBeOmittedWhenConditionOrChildStatementIsMultiLine` and start giving more warnings than it did in previous versions?

**Comment by sharwell** [comment]
@jnm2 Yes that's correct

**Comment by jnm2** [comment]
Hmm, the term 'condition' isn't generic enough since we have to handle `using` and `for` statements.

**Comment by jnm2** [comment]
`SA1519BracesMustNotBeOmittedWhenParentOrChildPartsAreMultiLine`?
